eclair-core/src/main/scala/fr/acinq/eclair/crypto/KeyManager.scala

```diff
@@ -54,77 +54,78 @@ trait KeyManager {
   }
 
   /**
-   *
    * @param isFunder true if we're funding this channel
    * @return a partial key path for a new funding public key. This key path will be extended:
    *         - with a specific "chain" prefix
    *         - with a specific "funding pubkey" suffix
    */
-  def newFundingKeyPath(isFunder: Boolean) : DeterministicWallet.KeyPath
+  def newFundingKeyPath(isFunder: Boolean): DeterministicWallet.KeyPath
 
   /**
-    *
-    * @param tx        input transaction
-    * @param publicKey extended public key
-    * @return a signature generated with the private key that matches the input
-    *         extended public key
-    */
+   * @param tx        input transaction
+   * @param publicKey extended public key
+   * @return a signature generated with the private key that matches the input
+   *         extended public key
+   */
   def sign(tx: TransactionWithInputInfo, publicKey: ExtendedPublicKey): ByteVector64
 
   /**
-    * This method is used to spend funds send to htlc keys/delayed keys
-    *
-    * @param tx          input transaction
-    * @param publicKey   extended public key
-    * @param remotePoint remote point
-    * @return a signature generated with a private key generated from the input keys's matching
-    *         private key and the remote point.
-    */
-  def sign(tx: TransactionWithInputInfo, publicKey: ExtendedPublicKey, remotePoint: PublicKey): ByteVector64
+   * This method is used to spend funds sent to htlc keys/delayed keys
+   *
+   * @param tx          input transaction
+   * @param publicKey   extended public key
+   * @param remotePoint remote point
+   * @param sighashType sighash flags
+   * @return a signature generated with a private key generated from the input keys's matching
+   *         private key and the remote point.
+   */
+  def sign(tx: TransactionWithInputInfo, publicKey: ExtendedPublicKey, remotePoint: PublicKey, sighashType: Int): ByteVector64
```

Comment:
I'm not a huge fan of directly exposing the `sighash` here, I would have liked a slightly higher-level abstraction.
If you have a suggestion, I'd love to hear it! The gist of it is that we need to distinguish signing local vs remote txs: when signing local txs we will always use `SIGHASH_ALL`, but when signing remote txs we may use less restrictive sighash flags (in anchor outputs, we use `SIGHASH_SINGLE | SIGHASH_ANYONECANPAY` for htlc txs.

Reply:
I'd like the sighash flag to be somehow encapsulated in the `TransactionWithInputInfo`.
One thing I can try, is to enrich `TransactionWithInputInfo` with either `Local` or `Remote` -> this way our type system will let us know whether this is a local tx or a remote tx, which could be useful.
Then `TransactionWithInputInfo` can expose a method that returns the sighash that should be used depending on the commitment format. I'll experiment with that and will see if it's better.

Reply:
The sighash handling feels a bit clunky, but it's not too bad either and I don't have a better proposition.

Reply:
Yeah, I think we can't fully hide it as the signing interface should allow setting it explicitly (it's part of a signature's "settings").
For now it's not used that much since only anchor output HTLCs use something different than `sighash_all`, but if a later commitment format makes more extensive use of different flags we may want to revisit that to ensure the code stays simple